UNRELEASED.md

```diff
@@ -6,6 +6,9 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Enhancements
 
+- [Layout] Add `id` prop for hash linking to `AnnotatedSection` components
+- [Heading] Add `id` prop
```

Comment:
Can we change this to something like:
```suggestion
- Added `id` prop to `Layout` and `Heading` for hash linking ([#4307](https://github.com/Shopify/polaris-react/pull/4307))
```

To match our [guidelines](https://github.com/Shopify/polaris-react/blob/main/documentation/Versioning%20and%20changelog.md#user-content-component-name-formats)